### ISSUE TITLE
[bug] 

### ISSUE THREAD

**Issue by BeycanDeveloper** [WalletConnect/web3modal#1527]
### Link to minimal reproducible example

It's not necessary

### Summary

Hello, I use WalletConnect in my project for both Tron and EVM-based networks. In doing this, I use the following package provided by Tron.

[package](https://github.com/tronprotocol/tronwallet-adapter)

However, since they both try to define the same "web components", I get the following error.

DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "w3m-button" has already been used with this registry

Therefore, I think you can solve this problem by adding a small condition to the places where these components are defined.

### List of related npm package versions

"@wagmi/core": "^1.4.7",
"@web3modal/wagmi": "^3.3.2",
"viem": "^1.19.11"

Usingi n Tron WalletConnect adapter

"@tronweb3/walletconnect-tron": "2.0.0",
"@wagmi/core": "^1.1.0",
"@walletconnect/sign-client": "^2.1.4",
"@walletconnect/types": "^2.1.4", 
"@web3modal/ethereum": "^2.4.2",
"viem": "^0.3.50",

**Comment by glitch-txs** [comment]
This is not recommended as you might end up with localStorage issues. We're planning to add support for more chains in the future though.

**Comment by BeycanDeveloper** [comment]
I understand that I have moved the codes in the tron package to a separate area right now and I will try to replace them with WalletConnect 1.

**Comment by way2ex** [comment]
@glitch-txs  Hi~currently, the package `@web3modal/ethers` and `@web3modal/wagmi` can both only  be used for eip155 namespace. Is there any plan to support other namespace such as `tron:0x94a9059e` namespace ? Thank you!

**Comment by glitch-txs** [comment]
Hi we're working a Solana integration and are planning to add Tron in the future as well, there's no current support for other namespaces atm

**Comment by BeycanDeveloper** [comment]
@glitch-txs how about Bitcoin and maybe for Xverse wallet? WalletConnect really needed for Bitcoin. Also, i want create a open source project MultipleChain i will write with TS can we talk with you about of structure?

**Comment by glitch-txs** [comment]
could you open a separated issue for each of those chains?

**Comment by BeycanDeveloper** [comment]
@glitch-txs yes i can open issue for Bitcoin & Xverse wallet support. But i want talk with you about of a project struct. That's why i contacted with you on Twitter can we talk there?

**Comment by glitch-txs** [comment]
happy to take PRs @BeycanDeveloper but I think we are a bit early to support multi chain in Web3Modal

**Comment by way2ex** [comment]
> Hi we're working a Solana integration and are planning to add Tron in the future as well, there's no current support for other namespaces atm

Thanks a lot for your great work!

**Comment by Muzided** [comment]
i am facing the same issue in my project i have added web3 modal for USDT transfers for multiple mainnet evm chains but for tron i am using "@tronweb3/walletconnect-tron": "^2.0.0",  library for trust wallet and i am getting the same issue how can i differentiate the web component

**Comment by hhun** [comment]
> Hi we're working a Solana integration and are planning to add Tron in the future as well, there's no current support for other namespaces atm

@glitch-txs 
TRON now has a high market share in the cryptocurrency world, at least higher than Solana. It is recommended to increase support for the TRON network.

**Comment by BeycanDeveloper** [comment]
@hhun you can use legacy version with this package: https://github.com/MultipleChain/tron-walletconnect
